File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/daos/UserInformationDao.java
package mx.edu.upqroo.kristenandroid.data.database.daos;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import mx.edu.upqroo.kristenandroid.data.database.entities.UserInformation;

@Dao
public interface UserInformationDao {
    @Insert
    void insert(UserInformation profile);

    @Query("DELETE FROM user_information")
    void deleteAll();

    @Query("SELECT * FROM user_information WHERE userId = :id")
    LiveData<UserInformation> getUserInformation(@NonNull String id);

    @Query("SELECT * FROM user_information WHERE userId = :id")
    UserInformation getUserInformationNotLive(@NonNull String id);
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/data/database/daos/UserInformationDao.kt
package mx.edu.upqroo.kristenandroid.data.database.daos

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import mx.edu.upqroo.kristenandroid.data.database.entities.UserInformation

@Dao
interface UserInformationDao {
    @Insert
    fun insert(profile: UserInformation)

    @Query("DELETE FROM user_information")
    fun deleteAll()

    @Query("SELECT * FROM user_information WHERE userId = :id")
    fun getUserInformation(id: String): LiveData<UserInformation>

    @Query("SELECT * FROM user_information WHERE userId = :id")
    fun getUserInformationNotLive(id: String): UserInformation
}
